### PR TITLE
Active Storage, AWS, Profile pictures

### PR DESCRIPTION
**Pull request by Jaybraum**
## Description
  * Sets up active storage for saving files
  * Sets up AWS S3 bucket for Deployment
  * Adds profile picture attachment to user creation
  * Refactors user/users request tests
  * Configs Cors file for Front End

## Added Gems

* gem "aws-sdk-s3", require: false
* gem 'rack-cors'
* gem 'image_processing', '~> 1.2'
* gem 'active_storage_validations'
  
## Concerns

* Checking if a user has a profile picture attached / Can change to be mandatory 
* Setting up another bucket in AWS for testing environment

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactor (**Removed Profile picture from migration**)

## Checklist:
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Issues: 
- [x] Closes Issue #7 
- [x] Closes Issue #6

### CODE REVIEW

File: config/initializers/cors.rb [hunk 1]
```diff
@@ -5,12 +5,12 @@
...
-#
-#     resource '*',
-#       headers: :any,
-#       methods: [:get, :post, :put, :patch, :delete, :options, :head]
-#   end
-# end
+Rails.application.config.middleware.insert_before 0, Rack::Cors do
+  allow do
+    origins 'localhost:3000', 'https://elegy-backend.herokuapp.com/'
+
```
Comment (NoahZinter) [inline config/initializers/cors.rb:11]: Nice, this should allow us to receive requests from the front end without triggering Cors problems

File: config/routes.rb [hunk 2]
```diff
@@ -1,7 +1,9 @@
 Rails.application.routes.draw do
   namespace :api do
     namespace :v1 do
-      resources :users, only: [:create]
+      resources :users, only: [:create] do
+        get :profile_picture, on: :member
```
Comment (NoahZinter) [inline config/routes.rb:5]: This line 5 is syntax related to active storage?

File: db/migrate/20210830114756_remove_profile_picture_from_users.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class RemoveProfilePictureFromUsers < ActiveRecord::Migration[5.2]
+  def change
+    remove_column :users, :profile_picture, :string
+  end
```
Comment (NoahZinter) [inline db/migrate/20210830114756_remove_profile_picture_from_users.rb:4]: So we're taking the profile picture out of this table in order to store it with activestorage

File: db/seeds.rb [hunk 4]
```diff
@@ -8,4 +8,10 @@
 
 User.destroy_all
 
-User.create!(email: 'ex@ample.com', name: 'Elder Bobby', date_of_birth: '1999/01/02', obituary: 'Tedious and brief', password: 'password', profile_picture: 'https://http.cat/201')
\ No newline at end of file
+user_1 = User.create!(email: 'ex@ample.com', name: 'Elder Bobby', date_of_birth: '1999/01/02', obituary: 'Tedious and brief', password: 'password')
+
+user_1.profile_picture.attach(
```
Comment (NoahZinter) [inline db/seeds.rb:13]: Will this be the syntax for creating these activestorage dependent in the future?

File: spec/api/v1/requests/users_request_spec.rb [hunk 5]
```diff
@@ -31,15 +29,15 @@
...
-        profile_picture: 'www.img-example.com'
+        password: 'password'
       }
       headers = {"CONTENT_TYPE"  => 'application/json'}
       post '/api/v1/users', headers: headers, params: JSON.generate(user_details)
 
       serialized_user = JSON.parse(response.body, symbolize_names: true)
 
       expect(serialized_user).is_a? Hash
+      #binding.pry
```
Comment (NoahZinter) [inline spec/api/v1/requests/users_request_spec.rb:40]: Will you remove this pry before merging?

**Review by NoahZinter — APPROVED** [review]
Will you remove the

File: config/initializers/cors.rb [hunk 6]
```diff
@@ -5,12 +5,12 @@
...
-#     origins 'example.com'
-#
-#     resource '*',
-#       headers: :any,
-#       methods: [:get, :post, :put, :patch, :delete, :options, :head]
-#   end
-# end
+Rails.application.config.middleware.insert_before 0, Rack::Cors do
+  allow do
+    origins 'localhost:3000', 'https://elegy-backend.herokuapp.com/'
```
Comment (tylrs) [inline config/initializers/cors.rb:10]: FYI you will need to add our deployed front end here when we get it up!

File: app/controllers/api/v1/users_controller.rb [hunk 7]
```diff
@@ -11,9 +12,19 @@ def create
     end
   end
 
+  def profile_picture
+    user = User.find_by(id: params[:id])
+  
+    if user&.profile_picture&.attached?
```
Comment (tylrs) [inline app/controllers/api/v1/users_controller.rb:18]: We can have a default profile picture easily on the front end as well. Might be easier there?

File: spec/api/v1/requests/users_request_spec.rb [hunk 8]
```diff
@@ -97,8 +91,7 @@
         date_of_birth: '2001/02/03',
         name: 'Jane Doe',
         obituary: 'Tedious and brief',
-        password: 'password',
-        profile_picture: 'www.img-example.com'
+        password: 'password'
```
Comment (tylrs) [inline spec/api/v1/requests/users_request_spec.rb:93]: I think we might need a new password

**Review by tylrs — APPROVED** [review]

**Pull request merged**